layers/meta-resin-intel/recipes-core/initrdscripts/files/init-external-media.sh

```diff
@@ -43,14 +43,26 @@ read_args() {
     done
 }
 
-boot_external_rootfs() {
-    mkdir /.flasher_root
-    mount /dev/disk/by-label/flash-root /.flasher_root
+boot_rootfs() {
+   local _what=$1
+
+    echo "init-external-media.sh: Waiting for udev to populate /dev/disk/by-label/$_what... "
+    while true
+    do
+        if ls -A /dev/disk/by-label/$_what >/dev/null 2>&1; then
+            break
+        fi
+        sleep 1
+    done
+    echo "init-external-media.sh: Found $_what label."
+
+    mkdir /$_what
+    mount /dev/disk/by-label/$_what /$_what
     # Watches the udev event queue, and exits if all current events are handled
     killall "${_UDEV_DAEMON##*/}" 2>/dev/null
 
-    exec switch_root /.flasher_root /sbin/init ||
-        fatal "Couldn't switch_root, dropping to shell"
+    exec switch_root /$_what /sbin/init ||
+        fatal "init-external-media.sh: Couldn't switch_root, dropping to shell"
```

Comment:
Actually, I don't think this code is valid at all. The special builtin exec doesn't return to the shell: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_20
"If command is specified, exec shall not return to the shell;"